cmd/notary/repo_factory.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"github.com/spf13/viper"
+
+	"github.com/docker/notary"
+	"github.com/docker/notary/client"
+	"github.com/docker/notary/client_api/api"
+	"github.com/docker/notary/tuf/data"
+	"github.com/docker/notary/utils"
+	"net/http"
+)
+
+const remoteConfigField = "api"
+
+type repoFactory func(gun data.GUN) (client.Repository, error)
```

Comment:
I think the naming here could be confusing, because `RepoBuilder` already exists. Maybe more explicitly `clientRepoFactory`? (same for filename as well)